src/modules/Elsa.Retention/Jobs/CleanupJob.cs

```diff
@@ -71,7 +71,14 @@ public async Task ExecuteAsync(CancellationToken cancellationToken = default)
 
                     await foreach (var entities in collector.GetRelatedEntitiesGeneric(page.Items).WithCancellation(cancellationToken))
                     {
-                        await cleanupService.Cleanup(entities);
+                        try
+                        {
+                            await cleanupService.Cleanup(entities);
+                        }
+                        catch (Exception exp)
+                        {
+                            _logger.LogError(exp, "Failed to clean up {Type} because exception thrown", collectorService.Key.Name);
```

Comment:
This fix only prevents the CleanupJob from crashing the entire Elsa server should the Cleanup service throw an exception.  However it does not fix the problem where a bad WorkflowIhstance (with undeserializable data) cannot be deleted. The above fix in WorkflowInstanceStore is required for that.